### PR TITLE
Reduce dependabot spam

### PR DESCRIPTION
**Pull request by abottoms-coder**
### Summary of Changes
Closes issue #1209 

### How to Test
1. Not actually testable. I could not get their "dry-run" ruby script/docker solution to actually work.

> *Demo GIF(s) and screenshots for testing procedure*

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [ ] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [ ] Check against the criteria:

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [ ] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -14,6 +14,21 @@ updates:
...
+      - dependency-name: "redux*"
+        update-types: ["version-update:semver-patch"]
+      - dependency-name: "cypress*"
+        update-types: ["version-update:semver-patch"]
+      - dependency-name: "prettier*"
+        update-types: ["version-update:semver-patch"]
+      - dependency-name: "eslint*"
+        update-types: ["version-update:semver-patch"]
+      - dependency-name: "react-dom*"
+        update-types: ["version-update:semver-patch"]
```
Comment (jorgegonzalez) [inline .github/dependabot.yml:27]: Nit, but `react-dom` shouldn't be ignored here, in my opinion.

Comment (abottoms-coder) [reply .github/dependabot.yml:27]: Thanks, will change

File: .github/dependabot.yml [hunk 2]
```diff
@@ -14,6 +14,19 @@ updates:
       - "raft review"
     reviewers:
       - "raft-tech/tdrs-developers"
+    ignore:
+      - dependency-name: "redux*"
+        update-types: ["version-update:semver-patch"]
```
Comment (jtwillis92) [inline .github/dependabot.yml:19]: Might be useful to add a link to [this article](https://github.blog/changelog/2021-05-21-dependabot-version-updates-can-now-ignore-major-minor-patch-releases/) if this file accepts comments.

**Review by jtwillis92 — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (abottoms-coder) [reply .github/dependabot.yml:19]: Ok, will add a standard yaml comment. Hopefully, dependabot takes it.

**Comment by abottoms-coder** [comment]
Don't merge until Monday :) @abottoms-coder

File: docs/Architecture Decision Record/016-dependabot-dependency-management.md [hunk 3]
```diff
@@ -48,3 +48,7 @@ Dev dependencies are listed in two locations in this project:
 Time estimate for development: n/a.
 
 [#917](https://github.com/raft-tech/TANF-app/pull/917) and [#932](https://github.com/raft-tech/TANF-app/pull/932) were created to support this migration and PR [#944](https://github.com/raft-tech/TANF-app/pull/944) completes this migration.
+
+### Addendum
+
+We are seeing a plethora of recurring updates so to stem the tide here, we are going to decrease dependabot's sensitivity to patch-level updates and for the worst offenders we will only be updating on minor version updates. This is denoted in our [dependabot config](https://github.com/raft-tech/TANF-app/.github/dependabot.yml).
```
Comment (ADPennington) [inline docs/Architecture Decision Record/016-dependabot-dependency-management.md:54]:
```suggestion
We are seeing a plethora of recurring updates so to stem the tide here, we are going to decrease dependabot's sensitivity to patch-level updates and for the worst offenders we will only be updating on minor version updates. This is denoted in our [dependabot config](https://github.com/raft-tech/TANF-app/.github/dependabot.yml). Note that this change will not prevent security-related patch updates (to the extent that dependency managers identify these updates as such). 
```

**Review by ADPennington — APPROVED** [review]
👍🏾 ticked off relevant items in PR summary (this is primarily a config change that will be tested upon merge).

**Pull request merged**